### ISSUE TITLE
Question on how to reconstruct images

### ISSUE THREAD

**Issue by MrHuff** [rosinality/glow-pytorch#7]
Hi!

Thanks a lot for this repo, it's really great! I am wondering how to reconstruct images, since the current reverse method seems to take a list of images with different sizes.

I tried writing a method:

```
    def reverse_data(self,z):
        for i, block in enumerate(self.blocks[::-1]):
            if i == 0:
                eps = torch.randn_like(z)
                input = block.reverse(z,eps)

            else:
                eps = torch.zeros_like(input)
                input = block.reverse(input,eps)

        return input
```
But I am unsure whether this is correct. I would appreciate any help!

Thank you!

**Comment by rosinality** [comment]
First you need to gather zs from blocks. Then you can reverse using these zs, but you will need to skip sampling (reparameterization) using prior. You can may refer to this: 4beed5d8837fb3208fc8ae8d8526db534b260dc1.

**Comment by MrHuff** [comment]
Hi I just tried it out and it works perfectly now! Thank you very much your help!

**Comment by koriavinash1** [comment]
for reconstruction do we need to normalize zs (obtained during the forward pass)? 

Currently, I'm training glow on mnist with affine coupling, during the forward pass zs blows up to 100k sometimes, due to which reconstructions are Nan always. Let me if normalization is required or if I'm missing any other step during reconstruction.

Thank you!

**Comment by rosinality** [comment]
@koriavinash1 No, you don't need a normalization. It should be enough to give zs from forward to reverse with reconstruct=True.

**Comment by koriavinash1** [comment]
@rosinality thanks for the quick response, do you have any idea on how to resolve the NaN issue?

**Comment by rosinality** [comment]
@koriavinash1 You can try to reduce --n_flow and --n_block to make the model more stable.
